### DISCUSSION TITLE
FavoriteButton not working correctly

### DISCUSSION THREAD

**Discussion by StephanLippert** [neuefische/questions#796] [Web]
[https://github.com/gwittm/Digital-Recipe-Book/tree/US12-FavoritePage-Stephan](https://github.com/gwittm/Digital-Recipe-Book/tree/US12-FavoritePage-Stephan)

Hi! We try to implement a favorite button, that works at our overview page, the details page and at the favorite page. We implemented the Component with its functions. We can toggle it at every page but the state will not be saved and displayed at other pages. We appreciate any help!!!

**Comment by NikRoe** [comment]
Hi Stephan,

what are you aiming for exactly? Do you want the favourite state to be part of the data in your database or do you want this state not to be connected to your database?
I am asking because the previous state of this user story went in the direction of making an update to the recipe in the database and toggling isFavourite. 

Your question, however, suggests that you want it not to be connected to the database.

Can you please clarify this for me, since my answer depends heavily on the outcome you want to achieve.

Thanks in advance.

**Comment by gwittm** [comment]
Hi Niklas!
Sorry for the confusion. We want to add the favorite-State to our database.
